src/main/webapp/app/shared/sidebar/sidebar.component.ts

```diff
@@ -109,4 +128,44 @@ export class SidebarComponent implements OnDestroy, OnChanges, OnInit {
         };
         return this.sidebarData.sidebarType ? size[this.sidebarData.sidebarType] : 'M';
     }
+
+    openFilterExercisesDialog() {
+        this.initializeFilterOptions();
+
+        if (!this.sidebarDataBeforeFiltering) {
+            this.sidebarDataBeforeFiltering = cloneDeep(this.sidebarData);
+        }
+
+        this.modalRef = this.modalService.open(ExerciseFilterModalComponent, {
+            size: 'lg',
+            backdrop: 'static',
+            animation: true,
+        });
+
+        this.modalRef.componentInstance.sidebarData = cloneDeep(this.sidebarDataBeforeFiltering);
+        this.modalRef.componentInstance.exerciseFilters = cloneDeep(this.exerciseFilters);
+
+        this.filterSubscription?.unsubscribe();
```

Comment:
Do you need to unsubscribe here?

Reply:
In this particular case it should not be needed since the subscription should be directly tied to the lifetime of the modal 👍